ibm/data_source_ibm_container_cluster.go

```diff
@@ -352,7 +359,8 @@ func dataSourceIBMContainerClusterRead(d *schema.ResourceData, meta interface{})
 	for i, worker := range workerFields {
 		workers[i] = worker.ID
 	}
-	servicesBoundToCluster, err := csAPI.ListServicesBoundToCluster(name, "", targetEnv)
+	boundedServicesNamespaceID := d.Get("bounded_services_namespace_id").(string)
```

Comment:
Since its an optional fields use d.GetOk
```
var namespaceID string
if nID, ok := d.GetOk("bounded_services_namespace_id"); ok {
namespaceID = nID.(string)
}
	servicesBoundToCluster, err := csAPI.ListServicesBoundToCluster(name, namespaceID, targetEnv)
	if err != nil {
		return fmt.Errorf("Error retrieving services bound to cluster: %s", err)
	}
```